### PR TITLE
Create new opentelemetry-jaeger-propagator library

### PR DESCRIPTION
**Pull request by mattbodd**
## Description
Create `opentelemetry-jaeger-propagator` to prepare for `opentelemetry-jaeger` exporter deprecation (https://opentelemetry.io/blog/2022/jaeger-native-otlp/).

This is the first step taken from https://github.com/open-telemetry/opentelemetry-rust/issues/995#issuecomment-1851173398 towards resolving https://github.com/open-telemetry/opentelemetry-rust/issues/995.

## Changes

* Create a new `opentelemetry-jaeger-propagator` library
* Moves `opentelemetry-jaeger::Propagator` module to `opentelemetry-jaeger-propagator`

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by mattbodd** [comment]
@cijothomas please let me know if this is what you had in mind for https://github.com/open-telemetry/opentelemetry-rust/issues/995#issuecomment-1851173398. I would especially be interested in getting feedback on the style/approach I've taken (ie: need more docs, additional unit tests are too granular).

File: opentelemetry-jaeger-propagator/CHANGELOG.md
Comment (cijothomas) [inline opentelemetry-jaeger-propagator/CHANGELOG.md:9]: Wording suggestion: Please add something to the tune of
_This is a new create containing just the propagator functionality that was originally part of opentelemetry-jaeger._

File: opentelemetry-jaeger/CHANGELOG.md
Comment (cijothomas) [inline opentelemetry-jaeger/CHANGELOG.md:8]: lets be more descriptive here to smoothen migration.
This crate previously contained two functionalities - an exporter to Jaeger, and a Jaeger propagator. Going forward, Jaeger propagator is shipped as new crate <crate-name>. The JaegerExporter is deprecated, and users should use OTLP exporter instead. See <linktoexample> showing OTLPExporter with Jaeger.

File: opentelemetry-jaeger/README.md
Comment (cijothomas) [inline opentelemetry-jaeger/README.md:44]: this readme file needs to updated to reflect it no longer has propagator and warning about exporter being removed in the next release.

File: opentelemetry-jaeger-propagator/src/lib.rs
Comment (TommyCpp) [inline opentelemetry-jaeger-propagator/src/lib.rs:3]: nit: remove this

File: opentelemetry-jaeger-propagator/src/lib.rs
Comment (TommyCpp) [inline opentelemetry-jaeger-propagator/src/lib.rs:55]: Let's have a seprate file for this?

File: opentelemetry-jaeger-propagator/Cargo.toml
Comment (TommyCpp) [inline opentelemetry-jaeger-propagator/Cargo.toml:37]: remove those two and add `integration_test` feature

File: opentelemetry-jaeger/CHANGELOG.md
Comment (TommyCpp) [reply opentelemetry-jaeger/CHANGELOG.md:8]: > The JaegerExporter will be deprecated after the next release, and users should use OTLP exporter instead. See "linktoexample" showing OTLPExporter with Jaeger.

+1. Let's just call out this create will be deprecate with only exporter in it.

File: opentelemetry-jaeger-propagator/Cargo.toml
Comment (mattbodd) [reply opentelemetry-jaeger-propagator/Cargo.toml]: I replaced the full and default features with integration_test which is also empty right now. I kept the integration tests in opentelemetry-jaeger instead of migrating them over so I don't have any integration tests in opentelemetry-jaeger-propagator. Does this make sense and is this what you expect?

File: opentelemetry-jaeger-propagator/README.md
Comment (cijothomas) [inline opentelemetry-jaeger-propagator/README.md:7]:
```suggestion
[`Jaeger`] propagator integration for applications instrumented with [`OpenTelemetry`]. To export telemetry to Jaeger, use the opentelemetry-otlp crate.
```

File: opentelemetry-jaeger/README.md
Comment (cijothomas) [inline opentelemetry-jaeger/README.md:16]: https://github.com/open-telemetry/opentelemetry-rust/tree/main/examples/tracing-jaeger Add link to this example here.

File: opentelemetry-jaeger/README.md
Comment (cijothomas) [reply opentelemetry-jaeger/README.md:49]: https://github.com/open-telemetry/opentelemetry-rust/pull/1487/files#diff-d153afbd7102b8218509b0ce0b654c51ba1b8fee4c558c097f01226743a89870R7 need a small fix to remove propagator wording

File: opentelemetry-jaeger/README.md
Comment (mattbodd) [reply opentelemetry-jaeger/README.md]: Thanks, I added this example to the README.

**Review by cijothomas — APPROVED** [review]
Thanks.
We may need some more cleanups to make sure no leftovers exist that can cause confusion.

**Comment by mattbodd** [comment]
Thanks @cijothomas for the review and approval. Do we want to wait for TommyCpp and/or others to give a final review + approval before merging or is your approval enough.

**Comment by cijothomas** [comment]
> Thanks @cijothomas for the review and approval. Do we want to wait for TommyCpp and/or others to give a final review + approval before merging or is your approval enough.

For trivial changes, one approval is good. This needs one more, to catch things one person may have missed! Lets wait for @TommyCpp to review.

**Review by TommyCpp — APPROVED** [review]
LGTM. Thanks for working on this

**Pull request merged**
